rfc/java11.md

```diff
@@ -0,0 +1,173 @@
+| **Author(s)**  | Michael Plump ([`@plumpy`](https://github.com/plumpy)) |
+| **SIG / WG**   | Platform SIG |
+
+## Overview
+
+To keep current with the evolution of Java, Spinnaker needs to move from Java 8
+to Java 11. This RFC documents a process and timeline for doing so.
+
+### Goals and Non-Goals
+
+Goals:
+
+*   Provide a process for moving official Spinnaker builds to the Java 11 JRE
+*   Discuss ways we will encourage people using non-offical builds to also move
+    to the Java 11 JRE
+*   Provide a timeline for the final switch to a Java 11 JDK along with criteria
+    to decide if the timeline should be delayed
+
+Non-Goals:
+
+*   Providing a rationale for upgrading Java; all active Java projects need to
+    update their JDK version eventually as libraries we depend upon will also
+    depend on new versions of the JDK
+*   Debating which Java version we should upgrade to; we take it as a given that
+    only LTS releases are acceptable for most organizations, especially for
+    running critical infrastructure like Spinnaker
+*   Documenting and solving technical issues that present themselves in the
+    migration; those will be handled separately, but this document concerns
+    itself primarily with the larger processes and timeline
+*   Detailing the required release system changes; since the releases are still
+    entirely maintained by Google, these changes aren't particularly relevant to
+    the wider community
+
+### JRE vs. JDK
+
+There are two components of Java of interest to this RFC. The Java Runtime
+Environment (JRE) is the `java` binary and associated libraries used to _run_
+Spinnaker services. The Java Development Kit (JDK) is the `javac` binary and
+associated libraries used to _compile_ Spinnaker.
+
+Once we upgrade to the Java 11 JDK, Spinnaker will no longer be able to run in a
+Java 8 JRE. So the first (more complicated) step is to get everyone migrated to
+a Java 11 JRE. Only after that is done can we switch to the Java 11 JDK.
+
+The Java 11 JRE has some fairly large changes, most notably the deprecation of
+the long-standing CMS garbage collector. Large organizations with highly-tuned
+JVMs have shown hesitancy to upgrade until the new runtime has undergone
+extensive testing. (Both Google and Netflix are in this category.)
+
+For official Spinnaker containers, we control the JRE used to run Spinnaker. For
+those releases, this proposal takes a relatively slow ramp to the Java 11 JRE
+and provides an escape hatch back to Java 8 for users who experience problems.
+
+There are also some unknown number of customers who aren't using the Spinnaker-
+provided containers. We will need to do some outreach to make sure these users
+convert to a Java 11 JRE too.
+
+## Timeline
+
+### 1.18 (early January)
+
+For our container builds, we'll first move just two (of approximately ten)
+services to the new JRE. The two services will be `igor` and `orca`. By
```

Comment:
Any reason you picked these two? Igor I can see as not "everyone" will be using it, but orca is pretty core to spinnaker and typically something everyone is running (even if they aren't running every microservice). Maybe `igor` and `echo` would be the easiest?

Reply:
I guess it disappeared from the diff view because I pushed a new commit, but there was more discussion about this here: https://github.com/spinnaker/community/pull/62#discussion_r345726643

(The services chosen are definitely very much up for discussion.)

Reply:
We also have `kork` (significantly) and `fiat` (its client lib) exporting dependencies into other projects. There might be a step 0 here that ensures that both of those are compatible with both JRE 8 and JRE 11.